chain/ethereum/src/runtime/abi.rs

```diff
@@ -227,6 +300,77 @@ impl AscIndexId for AscEthereumEvent<AscEthereumTransaction_0_0_6, AscEthereumBl
     const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::EthereumEvent;
 }
 
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscEthereumLog {
+    pub address: AscPtr<AscAddress>,
+    pub topics: AscPtr<AscTopicArray>,
+    pub data: AscPtr<Uint8Array>,
+    pub block_hash: AscPtr<AscH256>,
+    pub block_number: AscPtr<AscH256>,
+    pub transaction_hash: AscPtr<AscH256>,
+    pub transaction_index: AscPtr<AscBigInt>,
+    pub log_index: AscPtr<AscBigInt>,
+    pub transaction_log_index: AscPtr<AscBigInt>,
+    pub log_type: AscPtr<AscString>,
+    pub removed: AscPtr<AscWrapped<bool>>,
+}
+
+impl AscIndexId for AscEthereumLog {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::Log;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscEthereumTransactionReceipt {
+    pub transaction_hash: AscPtr<AscH256>,
+    pub transaction_index: AscPtr<AscBigInt>,
+    pub block_hash: AscPtr<AscH256>,
+    pub block_number: AscPtr<AscBigInt>,
+    pub cumulative_gas_used: AscPtr<AscBigInt>,
+    pub gas_used: AscPtr<AscBigInt>,
+    pub contract_address: AscPtr<AscAddress>,
+    pub logs: AscPtr<AscLogArray>,
+    pub status: AscPtr<AscBigInt>,
+    pub root: AscPtr<AscH256>,
+    pub logs_bloom: AscPtr<AscH2048>,
+}
+
+impl AscIndexId for AscEthereumTransactionReceipt {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TransactionReceipt;
+}
+
+/// Introduced in API Version 0.0.7, this is the same as [`AscEthereumEvent`] with an added
+/// `receipt` field.
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscEthereumEvent_0_0_7<T, B>
+where
+    T: AscType,
+    B: AscType,
+{
+    pub address: AscPtr<AscAddress>,
+    pub log_index: AscPtr<AscBigInt>,
+    pub transaction_log_index: AscPtr<AscBigInt>,
+    pub log_type: AscPtr<AscString>,
+    pub block: AscPtr<B>,
+    pub transaction: AscPtr<T>,
+    pub params: AscPtr<AscLogParamArray>,
+    pub receipt: AscPtr<AscEthereumTransactionReceipt>,
+}
+
+impl AscIndexId for AscEthereumEvent_0_0_7<AscEthereumTransaction_0_0_1, AscEthereumBlock> {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::EthereumEvent;
+}
+
+impl AscIndexId for AscEthereumEvent_0_0_7<AscEthereumTransaction_0_0_2, AscEthereumBlock> {
```

Comment:
Only now I realize, why are we implementing `0.0.7` events for old Transaction types? 🤔 

I might be missing something, but this seems wrong.

Reply:
You are right. We only need it for the latest block and transaction type versions.

Removed them in 1fae709ee.